fix: update package manifest and other files to latest upstream version

ndom91 opened the pull request:
Hey @honnip, this is finally ready to go! Haha

You can check out the latest published version [here](https://github.com/ndom91/gitbutler-flathub/actions/runs/13010957054), I had added a GHA here that builds the flatpak and puts up a `.flatpak` artifact to download in order to test it.

The high-level changes include:

- Compile gitbutler in the flatpak build instead of using the `.deb` artifact from us
- Don't include `libwebkit2gtk-4.0` in the modules
- Compile `git` in the flatpak build
- Include 1 source code patch to swap out a `keyring` crate feature in order to use a different Linux keyring implementation (`linux-native` vs `sync-secret-service`), this fixes the bug regarding saving credentials in the OS keystore from the flatpak
- Include `aarch64` in the `flathub.json` architectures
- Add a `justfile` (task runner) and tasks for installing dependencies, setting up node and cargo sources, building the flatpak, etc.
- Update `README.md` to include more instructions for manually building, etc.
- Flatpak `finish-args` for:
  - Interacting with keyring and git
  - `XCURSOR` path updates in the flatpak env for ensuring custom system cursors work
  - Access to some additional host paths for `gnupg`, git config, and gitbutler config to ensure settings can be persisted. 
  
**Important question** - is it possible for us to change the package id from `com.gitbutler.gitbutler` to `com.gitbutler.app`? This PR is still a mix of both, I'd really like to be able to move to `com.gitbutler.app`, but either way - depending on your answer I'll clean this PR up to be set on one only.

Other than that, if you have no further questions, I think once we get the package id thing cleared up and I clean this PR up, we should be good to go! :crossed_fingers:

honnip commented:
> is it possible for us to change the package id from com.gitbutler.gitbutler to com.gitbutler.app?

It may be needed to publish `com.gitbutler.app` then mark `com.gitbutler.gitbutler` as deprecated

ndom91 commented:
> > is it possible for us to change the package id from com.gitbutler.gitbutler to com.gitbutler.app?
> 
> It may be needed to publish `com.gitbutler.app` then mark `com.gitbutler.gitbutler` as deprecated

Sounds good. Let's just keep it at `com.gitbutler.gitbutler` for now then. I don't want to have folks not be able to update easily if we have to make a whole new package and deprecate the previous one.

honnip commented on flathub.json:
Sorry, I meant to say that this file can be deleted.

> If you build for both x86_64 and aarch64 you do not need a flathub.json file. There will be no new architecture add or removed on current runtimes, which mean that if that situation ever occurred, it would only happen when changing the runtime version in your package.

https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on

ndom91 replied:
Okay yeah no problem, I've deleted it :+1:

ndom91 commented:
Looks like its built successfully, however, using that install link from the bot above leads me to install basically the latest `main` from `com.gitbutler.gitbutler`. 

It reports as `0.13.8` instead of the `0.5.1013` I have in this PR. Does the flathub bot not actually build the PR contents? :thinking:

ndom91 commented:
Ugh okay so I've been running the build process having network access to download npm dependencies, but it seems you can't do that for official flathub packages. Looking at others, for exmaple [podman desktop](https://github.com/flathub/io.podman_desktop.PodmanDesktop/blob/56859156175b33c999644fbbaa5b8bcbae92044c/io.podman_desktop.PodmanDesktop.yml#L42), they seem to download the whole `node_modules` (pnpm store) from elsewhere as a source to the build process.

Is there a way to avoid this? 

I see other rust applications like [Halloy](https://github.com/flathub/org.squidowl.halloy/blob/9057218f0a855a906f74819db00f704c24d9ac09/org.squidowl.halloy.json#L31) seem to just be able to `cargo --offline fetch` and then `cargo --offline build` without haivng the rust sources downloaded from elsewhere as a module source :thinking:

honnip commented:
Isn't that what `node-sources.json` is for?

ndom91 commented:
Yeah I'd guess so, but then that fails with the following message:

```
]2;flatpak-builder: Cleanup GitButlerError: module GitButler: Can't specify build-args in sandboxed build
```

> Source: https://buildbot.flathub.org/#/builders/19/builds/21915

And the only build-args I have here is `--share=network`. Let me try again here without that build-arg, maybe I'm reading too much into it haha

honnip commented:
It failed because npm cache is a different format than the pnpm store, but I have no idea why npm couldn't find the caches too.

```
npm error code ENOTCACHED
npm error request to https://registry.npmjs.org/@eslint%2fjs failed: cache mode is 'only-if-cached' but no cached response is available.
```

(Ran with `npm_config_cache=/run/build/GitButler/flatpak-node/npm-cache`

ndom91 commented:
> It failed because npm cache is a different format than the pnpm store, but I have no idea why npm couldn't find the caches too.
> 
> ```
> npm error code ENOTCACHED
> npm error request to https://registry.npmjs.org/@eslint%2fjs failed: cache mode is 'only-if-cached' but no cached response is available.
> ```
> 
> (Ran with `npm_config_cache=/run/build/GitButler/flatpak-node/npm-cache`

Yeah so we use pnpm and the `flatpak-node-generator` only takes a `package-lock.json` as input, so we've got to convert it in the build process. I tried this [pnpm-lock-to-npm-lock](https://github.com/jakedoublev/pnpm-lock-to-npm-lock) package initially. But it seems to not have included everything from the pnpm lockfile in the package-lock. For example, using that helper's generated `package-lock.json` to then generate the `node-sources.json` results in `day-js` missing ([src](https://buildbot.flathub.org/#/builders/26/builds/21944)). 

I then tried to write the convertor myself, but that [run](https://buildbot.flathub.org/#/builders/19/builds/21944) is missing `undici-types@5.26.5`. It did include `undici-types@6.x.y`, but we've got an earlier version also in the pnpm lockfile, which isn't in the converted package-lock.json.

Anywya, long story short - I'm going to continue trying to get the manual convertor to work for now. Do you have any other ideas?

ndom91 commented:
Okay now I'm at a bit of a loss.. haha

I finally got the generated `package-lock.json` to cover all the packages we have in the various pnpm workspace pkgs (monorepo packages) as well as the packages we have defined only in the top-level `pnpm-workspace.yaml` (using pnpm catalogs or whatever they're called), and then generate a `node-sources.json` that covers everything.

But we still get this error that some package is missing ([src](https://buildbot.flathub.org/#/builders/19/builds/22049))

```log
Running: pnpm install --offline --frozen-lockfile
FB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/GitButler --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-x86_64-1/build/.flatpak-builder/build/GitButler-1 --bind-mount=/run/build/GitButler=/srv/buildbot/worker/build-x86_64-1/build/.flatpak-builder/build/GitButler-1 --build-dir=/run/build/GitButler --bind-mount=/run/ccache=/srv/buildbot/worker/build-x86_64-1/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1738682254 '--env=CFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer ' '--env=CXXFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=PUBLIC_FLATPAK_ID=com.gitbutler.app --env=TAURI_SIGNING_PRIVATE_KEY_PASSWORD=b --env=TAURI_SIGNING_PRIVATE_KEY=a --env=TURBO_TELEMETRY_DISABLED=1 --env=RUST_BACKTRACE=all --env=CARGO_HOME=/run/build/gitbutler/cargo --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin:/usr/lib/sdk/node22/bin:/usr/lib/sdk/rust-nightly/extra/sdk/rust-nightly/bin:/run/build/GitButler/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=64 /srv/buildbot/worker/build-x86_64-1/build/.flatpak-builder/rofiles/rofiles-IBCW4E /bin/sh -c 'pnpm install --offline --frozen-lockfile'' on host
Scope: all 5 workspace projects
Lockfile is up to date, resolution step is skipped
Progress: resolved 1, reused 0, downloaded 0, added 0
Packages: +1366
Progress: resolved 1, reused 0, downloaded 0, added 0
Progress: resolved 1366, reused 0, downloaded 0, added 0
 ERR_PNPM_NO_OFFLINE_TARBALL  A package is missing from the store but cannot download it in offline mode. The missing package may be downloaded from https://registry.npmjs.org/dayjs/-/dayjs-1.11.13.tgz.
Progress: resolved 1366, reused 0, downloaded 0, added 0
FB: host_command_exited_cb 3301028 256
]2;flatpak-builder: Cleanup GitButlerError: module GitButler: Child process exited with code 1
```

In fact, I can explicitly find that `dayjs@1.11.13` entry in the `package-lock.json` as well as `node-sources.json` :thinking: :thinking:

honnip commented:
Can we use the `sveltekit-build` artifact from CI? https://github.com/gitbutlerapp/gitbutler/actions/runs/13149373292

ndom91 commented:
@honnip thanks for the PR, I've merged it. 

So yeah I think we can use the pre-existing publish job's `sveltekit-build` step. The node/pnpm steps in there are already generating a pnpm cache by the name of `node-cache-Linux-x64-pnpm-ca14ef70df60425dc6feb0d8823e8381158d5c4a9f494b362fd980e07c9e3e8f` for example. However, I'm not sure exactly what the contents of that would be nor exactlyh how to access it. I also see in the GHA env vars that the `PNPM_HOME` is set to `/home/runner/setup-pnpm/node_modules/.bin`. I believe we should be able to pull out the pnpm store from somewhere in there, right? I cuold add a seprate "upload-artifact" step to upload that for us for consumption here.. :thinking:
